tests/e2e/bq-batch-retrieval.py

```diff
@@ -359,4 +372,15 @@ def test_no_max_age(client):
     output = feature_retrieval_job.to_dataframe()
     print(output.head())
 
-    assert output["entity_id"].to_list() == output["feature_value8"].to_list()
\ No newline at end of file
+    assert output["entity_id"].to_list() == output["feature_value8"].to_list()
+
+
+@pytest.mark.dataflow_runner
+def test_write_terminate_jobs(client):
+    ingest_jobs = client.list_ingest_jobs()
+    ingest_jobs = [client.list_ingest_jobs(job.id)[0].external_id for job in ingest_jobs if job.status == IngestionJobStatus.RUNNING]
+
+    cwd = os.getcwd()
+    with open(f"{cwd}/ingesting_jobs.txt", "w+") as output:
```

Comment:
This isn't really a test right?

How are you going to ensure that this code always runs at the end? I think it might be safer to use https://docs.pytest.org/en/latest/fixture.html#fixture-finalization-executing-teardown-code

You want to make sure that the ingestion_jobs.txt is not created until right at the end.

Reply:
Updated test with fixture for tearing down infra. It would only run for dataflow_runner marker.